test/Microsoft.AspNet.Mvc.ViewFeatures.Test/Rendering/HtmlHelperCheckboxTest.cs

```diff
@@ -649,6 +649,73 @@ public void CheckBoxFor_WithComplexExpressions_DoesNotUseValuesFromViewDataDicti
             Assert.Equal(expected, HtmlContentUtilities.HtmlContentToString(html));
         }
 
+        [Fact]
+        public void Checkbox_UsesSpecifiedExpression()
+        {
+            // Arrange
+            var helper = DefaultTemplatesUtilities.GetHtmlHelper();
+
+            // Act
+            var checkboxResult = helper.CheckBox("Property1");
+
+            // Assert
+            Assert.Equal(
+                "<input id=\"HtmlEncode[[Property1]]\" name=\"HtmlEncode[[Property1]]\" type=\"HtmlEncode[[checkbox]]\" value=\"HtmlEncode[[true]]\" />" +
+                "<input name=\"HtmlEncode[[Property1]]\" type=\"HtmlEncode[[hidden]]\" value=\"HtmlEncode[[false]]\" />",
+                HtmlContentUtilities.HtmlContentToString(checkboxResult));
+        }
+
+        [Fact]
+        public void Checkbox_UsesSpecifiedIsChecked()
+        {
+            // Arrange
+            var helper = DefaultTemplatesUtilities.GetHtmlHelper();
+
+            // Act
+            var checkboxResult = helper.CheckBox("Property1", isChecked: true);
+
+            // Assert
+            Assert.Equal(
+                "<input checked=\"HtmlEncode[[checked]]\" id=\"HtmlEncode[[Property1]]\" name=\"HtmlEncode[[Property1]]\" type=\"HtmlEncode[[checkbox]]\" value=\"HtmlEncode[[true]]\" />" +
+                "<input name=\"HtmlEncode[[Property1]]\" type=\"HtmlEncode[[hidden]]\" value=\"HtmlEncode[[false]]\" />",
+                HtmlContentUtilities.HtmlContentToString(checkboxResult));
+        }
+
+        [Fact]
+        public void Checkbox_UsesSpecifiedIsCheckedRegardlessOfExpressionValue()
+        {
+            // Arrange
+            var metadataProvider = new EmptyModelMetadataProvider();
+            var helper = DefaultTemplatesUtilities.GetHtmlHelper(new ViewDataDictionary<TestModel>(metadataProvider));
+            helper.ViewContext.ClientValidationEnabled = false;
+            helper.ViewData.Model = new TestModel { Property1 = true };
+
+            // Act
+            var checkboxResult = helper.CheckBox("Property1", isChecked: false);
+
+            // Assert
+            Assert.Equal(
+                "<input id=\"HtmlEncode[[Property1]]\" name=\"HtmlEncode[[Property1]]\" type=\"HtmlEncode[[checkbox]]\" value=\"HtmlEncode[[true]]\" />" +
+                "<input name=\"HtmlEncode[[Property1]]\" type=\"HtmlEncode[[hidden]]\" value=\"HtmlEncode[[false]]\" />",
+                HtmlContentUtilities.HtmlContentToString(checkboxResult));
+        }
+
+        [Fact]
+        public void Checkbox_UsesSpecifiedHtmlAttributes()
+        {
+            // Arrange
+            var helper = DefaultTemplatesUtilities.GetHtmlHelper();
+
+            // Act
+            var checkboxResult = helper.CheckBox("Property1", htmlAttributes: new { attr = "value" });
```

Comment:
Something more to track for another PR: `htmlAttributes` is another "value source". Not sure even the existing test classes cover including `checked`, `id`, `name`, `type` or `value` properties in the `htmlAttributes` object.

FYI the generated HTML might override a `name` and `value` (when the `value` argument is non-`null`) but `htmlAttributes` should "win" in general.